### ISSUE TITLE
Find out whether the ClusterRole works as expected

### ISSUE THREAD

**Issue by snizhana-dynnyk** [giantswarm/roadmap#641]
RBAC: ClusterRole `write-client-certificates` does not include permission to fetch the Secret. Name and permissions are not aligned.

**Comment by kuosandys** [comment]
The ClusterRole `write-client-certificates` was created in order to grant the `automation` ServiceAccount permissions to create WC client certificates ([related PR](https://github.com/giantswarm/rbac-operator/pull/204)). The ClusterRoleBinding `write-client-certificates-customer-sa` is created between this ClusterRole and this ServiceAccount.

The `automation` ServiceAccount also has a RoleBinding `write-all-customer-sa` to the `cluster-admin` ClusterRole. In the bootstrap package, this RoleBinding is created in the `default` namespace ([for reference](https://github.com/giantswarm/rbac-operator/blob/336f89e359b2276ae76902ec5d5358e5d8452416/service/internal/bootstrap/resources.go#L473)). Subjects bound to this Role are granted access for all verbs to all resources in all API groups, including `Secrets`.

However, it seems that in addition to the above, a RoleBinding between the `automation` ServiceAccount and the `cluster-admin` ClusterRole is also created in every organization namespace.

For example, the following commands both return `yes`:
`kubectl auth can-i get secrets --as system:serviceaccount:default:automation -n default`
`kubectl auth can-i get secrets --as system:serviceaccount:default:automation -n org-giantswarm`

(Small experiment: `rbac-operator` recreates the `write-all-customer-sa` RoleBinding within an organization namespace if you delete it. This can be observed from the logs of the `rbac-operator` pods.)

TL;DR The ClusterRole `write-client-certificates` does not include permissions to fetch `Secrets, but the `automation` ServiceAccount for which the ClusterRole was created for (and bound to), does have the permissions to fetch `Secrets`.

**Comment by marians** [comment]
We talked today. I think it will be part of a general default (cluster)role cleanup. Closing.
